### ISSUE TITLE
xlsx column format

### ISSUE THREAD

**Issue by farzeni** [jazzband/tablib#252]
Hello, 

thanks for your good work.

I have an issue exporting data in xlsx format. I have a data table with different type of columns: some are numerics (eg prices) and needs a numeric type column, others are in text format but with a numeric content (phone numbers with leading zeros).

When i try to export them, giving the right format to the value (char, int, float, etc..) it happens that cell values are converted without considering the initial format, the one i set.

This happens because tablib xlsx writer converts all values to unicode:

https://github.com/kennethreitz/tablib/blob/master/tablib/formats/_xlsx.py#L140

``` python
              try:
                if '\n' in col:
                        ws.cell('%s%s'%(col_idx, row_number)).value = unicode(
                            '%s' % col, errors='ignore')
                        style = ws.get_style('%s%s' % (col_idx, row_number))
                        style.alignment.wrap_text
                    else:
                        ws.cell('%s%s'%(col_idx, row_number)).value = unicode(
                            '%s' % col, errors='ignore')
                except TypeError:
                    ws.cell('%s%s'%(col_idx, row_number)).value = unicode(col)
```

so this cause openyxl to auto parse values with numeric content as a numeric cell:

https://github.com/kennethreitz/tablib/blob/master/tablib/packages/openpyxl3/cell.py#L219

``` python
       def data_type_for_value(self, value):
        """Given a value, infer the correct data type"""
        if value is None:
            data_type = self.TYPE_NULL
        elif value is True or value is False:
            data_type = self.TYPE_BOOL
        elif isinstance(value, (int, float)):
            data_type = self.TYPE_NUMERIC
        elif not value:
            data_type = self.TYPE_STRING
        elif isinstance(value, (datetime.datetime, datetime.date)):
            data_type = self.TYPE_NUMERIC
        elif isinstance(value, str) and value[0] == '=':
            data_type = self.TYPE_FORMULA
        elif self.RE_PATTERNS['numeric'].match(value):
            data_type = self.TYPE_NUMERIC
        elif value.strip() in self.ERROR_CODES:
            data_type = self.TYPE_ERROR
        else:
            data_type = self.TYPE_STRING
        return data_type
```

instead of converting all cell values to unicode,  wouldn't be possible to write cell values using openpyxl's 'set_value_explicit' checking the value format and passing the value using right openpyxl cell.TYPE_\* ?

**Comment by vToMy** [comment]
+1 I get this problem for date types as well.

**Comment by isedwards** [comment]
For other people struggling with this, it's worth noting that the problem only affects the `.xlsx` method that produces Excel output.  Therefore a very quick workaround in the meantime is to bypass the method and use `openpyxl` to write the file directly.  Something simple like...

    import openpyxl

    def write_xlsx(tablib_dataset, filename):
        wb = Workbook()
        ws = wb.active
        ws.title = "Sheet 1"

        # Write the header row and make cells bold
        ws.append(tablib_dataset.headers)
        for cell in ws[1:1]:
            cell.font = Font(bold=True)

        for row in tablib_dataset:
            ws.append(row)

        wb.save(filename=filename)

**Comment by claudep** [comment]
Fixed by 5fde5259d9adcfc47db93110d2dac6b08f31e24f
